src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.AutoreleasePool.Unsupported.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Versioning;
+
+namespace System.Threading
+{
+    internal sealed partial class ThreadPoolWorkQueue
+    {
+        [SupportedOSPlatform("macos")]
+        [SupportedOSPlatform("ios")]
+        [SupportedOSPlatform("tvos")]
+        [SupportedOSPlatform("watchos")]
+        private static void DispatchItemWithAutoreleasePool(object workItem, Thread currentThread)
```

Comment:
The call to this method is under `#if TARGET_OSX || TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS`. Is this dummy impl still needed?